source/_posts/routing-in-react.md

```diff
@@ -0,0 +1,387 @@
+---
+title: Routing in React 🚀
+contributor: Saurabh Ghatnekar
+avatar:
+description: Learn how to use React Router to create a single-page application with multiple routes.
+date: 2023-02-17
+tags: [react, routing]
+id: routing-in-react
+relatedPost: 
+bannerHeader: 
+bannerSubheader: 
+---
+Routing in React
+================
+## Table of Contents
+<!-- toc -->
+    I. Introduction 
+    	A. Definition of Routing 
+    II. React Router 
+    	A. Overview 
+    	B. Installation 
+    	C. Usage
+    III. React Router Components 
+    	A. BrowserRouter 
+    	B. Route 
+    	C. Link and NavLink 
+    	D. Nested Routes
+    IV. React Router with Redux 
+    	A. Overview 
+    	B. Usage
+    V. Conclusion 
+    	A. Frequently Encountered Issues
+    	B. Summary of Routing in React
+
+Introduction
+------------
+
+Routing in React JS is a way for a computer program to help you move between different pages or sections in a reactjs web application. Imagine that you have a special book that has many stories inside. Each story has its own page, and the book has a table of contents at the beginning that lists all the stories.
+
+When you want to read a different story, you can use the table of contents to find the page number for that story. Then you can turn to that page and start reading. This is similar to how routing works in a React app.
+
+The different stories in the book are like the different pages or components in a React app, and the table of contents is like a special tool that helps you move between them. When you click on a link in the table of contents, it takes you to the right page so you can see the story you want to read.
+
+React Router
+------------
+
+React Router enables "client-side routing". This module provides components to handle routing by defining the routes in your application and transitioning between them.
+
+Use create react app to quickly create a react app up and running.
+
+    npx create-react-app my-app
+
+To get started with React Router, you will first need to install the react router package. You can **install React Router** from the public npm registry with either npm or yarn in your react application. You can use the following command to install react router using npm.
+
+    npm install react-router-dom
+
+Once the library is installed, you can import the components you want to use in your application. For example, to use the **BrowserRouter**, **Route**, **Link,** and other components, you can add the following statement to your code to import them from the **React router library**
+
+    import { BrowserRouter, Route, Link } from 'react-router-dom';
```

Comment:
Script snippets should be highlighted. We use hljs for this. Please check for **example angular-architecture-best-practices.md** how it is done.